Read out layout from options directly, rather than via backend

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54208 Remove __get__ from Tensor stub.
* **#54074 Read out layout from options directly, rather than via backend**

I don't see why this shouldn't work.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27086594](https://our.internmc.facebook.com/intern/diff/D27086594)

ezyang committed:
Read out layout from options directly, rather than via backend

I don't see why this shouldn't work.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ljk53 commented on tools/codegen/api/python.py:
```diff
@@ -701,7 +701,7 @@ def signature(f: NativeFunction, *, method: bool = False, pyi: bool = False) ->
             type=OptionalType(BaseType(BaseTy.Layout)),
             default='strided' if pyi else 'torch.strided',
-            default_init='layout_from_backend(self.options().backend())' if is_like_or_new_function else None,
+            default_init='self.options().layout()' if is_like_or_new_function else None,
```
can it be 'self.layout()' directly?

ezyang replied:
Mmm, probably!

ezyang committed:
Update on "Read out layout from options directly, rather than via backend"


I don't see why this shouldn't work.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27086594](https://our.internmc.facebook.com/intern/diff/D27086594)

[ghstack-poisoned]
